src/Functions/FunctionsCoding.h

```diff
@@ -297,8 +297,18 @@ class FunctionIPv6StringToNum : public IFunction
                  out_offset < vec_res.size();
                  out_offset += IPV6_BINARY_LENGTH, ++i)
             {
+                auto src_string = std::string(reinterpret_cast<const char *>(&vec_src[src_offset]));
```

Comment:
I concerned about copying all data to temporary string. 
We can try to change code a bit. One possible solution is to allocate temporary buffer outside of loop, that contains `::ffff:` and then `IPV4_MAX_TEXT_LENGTH + 1` free bytes. If string is ipv4 then copy it to this buffer starting from position of this free bytes. Then pass buffer to parseIPv6 instead of column's data.

Also I'm not sure how this code with strings will be optimized, maybe most of overhead will be eliminated by compiler. 
It can be tested with performance test, but I don't see it for this function, you may write it.

Reply:
Sure, I have updated the code to not use `std::strings`.
Simply allocating a buffer and then using `strcat` for concatenating the IPv4 address.
Depending on the kind of IP just calling `parseIPv6` with suitable value.
Since I am no longer using strings, I think I will skip writing a performance test.